File: 2024/day3/part1.lua
dofile("UsefulStuff.lua")

local function part1(fileName)
	local total = 0
	for memory in io.lines(Script_path() .. fileName) do
		for a, b in memory:gmatch("mul%((%d+),(%d+)%)") do
			total = total + a * b
		end
	end
	return total
end

UnitTest(part1, "testcases/test1.txt", 161)
UnitTest(part1, "input.txt", 175615763)

local result = part1("input.txt")
print("Part 1: " .. tostring(result))

File: 2025/day3/Javascript/Part1.js
const { assert } = require('console');
const fs = require('fs');

function part1(filepath) {
    banks = fs.readFileSync(filepath, 'UTF8').toString()
    .replaceAll(/\r\n/g,"\n").split('\n')
    .map(x =>  x.split('').map(y => Number(y)))
    
    total = 0
    banks.forEach(bank => {
        first = Math.max(...bank.slice(0,-2))
        remaining = bank.slice(bank.indexOf(first)+1)
        second = Math.max(...remaining)
        total += first*10 + second
    });
    return total
}

assert(part1(__dirname + '/../testcases/test1.txt') == 357)
assert(part1(__dirname + '/../input.txt') == 17229)
console.log("Part 1:", part1(__dirname + '/../input.txt'))